### ISSUE TITLE
Countless environment bugs when running finetune.sh

### ISSUE THREAD

**Issue by JasonLeeUT** [2U1/Qwen2-VL-Finetune#90]
Any body succeeded in running the finetuning? Could you please show your environment configurations including GPUs, system, gcc , nvcc version? I have tried ubuntu and centos7, H20 and 4090, I have met countless bugs and I have spent four days for fixing them but it still doesn't work. Anybody succeeded?

**Comment by 2U1** [comment]
I wasn't having any bug when I was trying.
I'll take a look an let you know what is wrong.

**Comment by 2U1** [comment]
I've run a simple finetuning test with about 20 samples with mixed-modality and none mixed modality.
It didn't occur any bugs 

gpu spec

```
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.120                Driver Version: 550.120        CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA RTX A5000               Off |   00000000:31:00.0 Off |                  Off |
| 30%   30C    P8             23W /  230W |      10MiB /  24564MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA RTX A5000               Off |   00000000:4B:00.0 Off |                  Off |
| 30%   29C    P8             23W /  230W |      10MiB /  24564MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA RTX A5000               Off |   00000000:B1:00.0 Off |                  Off |
| 30%   30C    P8             22W /  230W |      10MiB /  24564MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA RTX A5000               Off |   00000000:CA:00.0 Off |                  Off |
| 30%   29C    P8             20W /  230W |      10MiB /  24564MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
+-----------------------------------------------------------------------------------------+

os/kernel version
```
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

Linux 75a5939453f0 6.8.0-49-generic #49~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Nov  6 17:42:15 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
```

gcc version
```
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

nvcc version
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Thu_Mar_28_02:18:24_PDT_2024
Cuda compilation tools, release 12.4, V12.4.131
Build cuda_12.4.r12.4/compiler.34097967_0
```

**Comment by 2U1** [comment]
I was using a docker and if you needed I'll make an image for it.

**Comment by 7998857** [comment]
Hey, I had some problems as well with the provided environment. Maybe it helps, if I share here, how I got it to work...
In my case it seemed to be related to the Cuda version which came with torch (too new maybe?). I tried just downgrading torch to 2.3 as it was suggested in some other thread, but it did not work since other packages then were incompatible.
I ended up installing the packages one after the other by hand via pip in a fresh venv, until the code ran without errors. This configuration seems to work now:

```
absl-py==2.2.1
accelerate==1.5.2
annotated-types==0.7.0
certifi==2025.1.31
charset-normalizer==3.4.1
deepspeed==0.15.1
einops==0.8.1
filelock==3.18.0
#flash-attn==2.6.3
fsspec==2025.3.0
grpcio==1.71.0
hjson==3.1.0
huggingface-hub==0.29.3
idna==3.10
Jinja2==3.1.6
liger_kernel==0.5.5
Markdown==3.7
MarkupSafe==3.0.2
mpmath==1.3.0
networkx==3.4.2
ninja==1.11.1.4
numpy==1.26.4
nvidia-cublas-cu12==12.4.5.8
nvidia-cuda-cupti-cu12==12.4.127
nvidia-cuda-nvrtc-cu12==12.4.127
nvidia-cuda-runtime-cu12==12.4.127
nvidia-cudnn-cu12==9.1.0.70
nvidia-cufft-cu12==11.2.1.3
nvidia-curand-cu12==10.3.5.147
nvidia-cusolver-cu12==11.6.1.9
nvidia-cusparse-cu12==12.3.1.170
nvidia-cusparselt-cu12==0.6.2
nvidia-nccl-cu12==2.21.5
nvidia-nvjitlink-cu12==12.4.127
nvidia-nvtx-cu12==12.4.127
packaging==24.2
peft==0.12.0
pillow==11.1.0
protobuf==6.30.2
psutil==7.0.0
py-cpuinfo==9.0.0
pydantic==2.11.1
pydantic_core==2.33.0
PyYAML==6.0.2
qwen-vl-utils==0.0.4
regex==2024.11.6
requests==2.32.3
safetensors==0.5.3
six==1.17.0
sympy==1.13.1
tensorboard==2.19.0
tensorboard-data-server==0.7.2
tokenizers==0.21.1
torch==2.6.0
torchvision==0.21.0
tqdm==4.67.1
transformers @ git+https://github.com/huggingface/transformers@b7fc2daf8b3fe783173c270d592073aabfb426cb
triton==3.2.0
typing-inspection==0.4.0
typing_extensions==4.13.0
ujson==5.10.0
urllib3==2.3.0
Werkzeug==3.1.3
```

where `flash-attn` has to be installed with no build isolation as described in the installation instructions:

`pip install flash-attn==2.6.3 --no-build-isolation`

With that venv the training scripts work. At least the `finetune_lora.sh` works. The normal one `finetune.sh` seems to run out of RAM for now.

Here is the rest of my setup

```
$ nvcc --version

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Aug_15_22:02:13_PDT_2023
Cuda compilation tools, release 12.2, V12.2.140
Build cuda_12.2.r12.2/compiler.33191640_0
```

```
$ gcc --version

gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
$ cat /etc/os-release

PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

```
$ nvidia-smi

+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 555.42.06              Driver Version: 555.42.06      CUDA Version: 12.5     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4090        Off |   00000000:01:00.0 Off |                  Off |
|  0%   45C    P0            244W /  450W |   10218MiB /  24564MiB |     62%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A     51381      C   ...wen2-VL-Finetune/venv_4/bin/python3      10180MiB |
+-----------------------------------------------------------------------------------------+
```

Actually, I think it may be running now due to the fact that the training script now uses the system Cuda version 12.2. With the conda setup it probably tried to use the Cuda that comes with torch. At least now I get the logs

```
Installed CUDA version 12.2 does not match the version torch was compiled with 12.4 but since the APIs are compatible, accepting this combination
```

But the training metrics look good.

**Comment by 2U1** [comment]
@JasonLeeUT @7998857 
I'll upload my docker image in the dockerhub.
I was using cuda 12.4 (via docker) and updated the flash-attn into the lastest one right now.

Maybe there are could be some conflicts in the env when I've updataed some libraires. 
I am using the torch2.5.1 cuda 12.4 version as the base image . It is a large one so it will take some time to upload.

Sorry again for the inconvinience.

**Comment by 2U1** [comment]
I've uploaded a pre-build image to simplfying the setup.

**Comment by 2U1** [comment]
I've uploaded the docker image and how to use it in the readme. Also mentioned about the system env.

Thanks for debugging!
